ivy/functional/ivy/general.py

```diff
@@ -2308,8 +2308,79 @@ def gather_nd(
     ret
         New array of given shape, with the values gathered at the indices.
 
+    Examples
+    --------
+    With :code:`ivy.Array` input:
+
+    >>> x = ivy.array([0., 1., 2., 3., 4., 5., 6.])
+    >>> y = ivy.array([1])
+    >>> print(ivy.gather_nd(x, y))
+    ivy.array(1.)
+
+    >>> x = ivy.array([[0., 1., 2.], \
+                       [3., 4., 5.]])
+    >>> y = ivy.array([[0], \
+                       [1]])
+    >>> z = ivy.array([[0., 0.], \
+                       [0., 0.]])
+    >>> ivy.gather_nd(x, y, device='cpu', out=z)
+    >>> print(z)
+    ivy.array([[0., 1.],
+               [2., 3.]])
+
+    >>> x = ivy.array([[[0., 1.], [2., 3.]], \
+                       [[4., 5.], [6., 7.]], \
+                       [[8., 9.], [10., 11.]]])
+    >>> y = ivy.array([[[0]], \
+                       [[1]], \
+                       [[2]]])
+    >>> ivy.gather_nd(x, y, out=x)
+    >>> print(x)
+    ivy.array([[[ 0.,  1.],
+                [ 2.,  3.]],
+               [[ 4.,  5.],
+                [ 6.,  7.]],
+               [[ 8.,  9.],
+                [10., 11.]]])
+
+    With :code:`ivy.NativeArray` input:
+
+    >>> x = ivy.native_array([0., 1., 2.])
+    >>> y = ivy.native_array([1])
+    >>> print(ivy.gather_nd(x, y))
+    ivy.array(1.)
+
+    With a mix of :code:`ivy.Array` and :code:`ivy.NativeArray` inputs:
+
+    >>> x = ivy.native_array([0., 1., 2.])
+    >>> y = ivy.array([1])
+    >>> print(ivy.gather_nd(x, y))
+    ivy.array(1.)
+
+    With a mix of :code:`ivy.Array` and :code:`ivy.Container` inputs:
+
+    >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), \
+                          b=ivy.array([4., 5., 6.]))
+    >>> y = ivy.array([1])
+    >>> print(ivy.gather_nd(x, y))
+    {
+        a: ivy.array(1.),
+        b: ivy.array(5.)
+    }
+
+    With :code:`ivy.Container` input:
+
+    >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), \
+                          b=ivy.array([4., 5., 6.]))
+    >>> y = ivy.Container(a=ivy.array([0]), \
+                          b=ivy.array([2]))
+    >>> print(ivy.gather_nd(x, y))
+    {
+     	a: ivy.array(0.),
```

Comment:
This line causes a lint failure as it seems to have a mix of spaces and tabs, please update that to have only spaces.